app/actions/ControlActions.js

```diff
@@ -485,6 +485,28 @@ export const revokeTicketsAttempt = (passphrase) => async (
   }
 };
 
+export const DISCOVERUSAGE_ATTEMPT = "DISCOVERUSAGE_ATTEMPT";
+export const DISCOVERUSAGE_FAILED = "DISCOVERUSAGE_FAILED";
+export const DISCOVERUSAGE_SUCCESS = "DISCOVERUSAGE_SUCCESS";
+
+export const discoverUsageAttempt = (gapLimit) => async (
+  dispatch,
+  getState
+) => {
+  dispatch({ type: DISCOVERUSAGE_ATTEMPT });
+  const walletService = sel.walletService(getState());
+  try {
+    const discoverUsageResponse = await wallet.discoverUsage(
+      walletService,
+      gapLimit
+    );
+    dispatch({ discoverUsageResponse, type: DISCOVERUSAGE_SUCCESS });
```

Comment:
After this line, add the following (before rescanning) to force reloading the address filters in SPV mode:

`await wallet.loadActiveDataFilters(walletService);`

This should fix the last issue with this PR.